File: 1-H/java/src/ExchangeHands.java

public class ExchangeHands {

    private static boolean hasused = false;
    private final Player exchanger;
    private final Player exchangee;
    private final int roundleft = 3;

    public ExchangeHands(Player exchanger, Player exchangee) {
        this.exchanger = exchanger;
        this.exchangee = exchangee;
    }

    public boolean hasUsed()  {
        return hasused;
    }

    public int getRoundleft() {
        return roundleft;
    }
}

File: 1-H/Python/exchangeHands.py
class ExchangeHands:
    def __init__(self, exchanger, exchangee):
        self.__exchanger = exchanger
        self.__exchangee = exchangee
        self.__roundleft = 3

    def exchange(self):
        hand = self.__exchanger.getHand
        self.__exchanger.setHand = self.__exchangee.getHand
        self.__exchangee.setHand = hand

    def countDown(self):
       self.__roundleft -= 1
       if self.__roundleft == 0:
           self.exchange()